Nos/Views/Components/FlagOptionPicker.swift

```diff
@@ -0,0 +1,105 @@
+import SwiftUI
+/// Displays a list of selectable flag options
+struct FlagOptionPicker: View {
+    @Binding private var selectedOption: FlagOption?
+    var options: [FlagOption]
+    var title: String
+    var subtitle: String?
+
+    init(selectedOption: Binding<FlagOption?>, options: [FlagOption], title: String, subtitle: String?) {
+        self._selectedOption = selectedOption
+        self.options = options
+        self.title = title
+        self.subtitle = subtitle
+    }
+
+    var body: some View {
+        VStack(alignment: .leading) {
+            HeaderView(text: title)
+            
+            if let subtitle = subtitle {
+                HeaderView(text: subtitle)
+            }
+            flagOptionsListView
+        }
+        .padding()
+    }
+
+    private var flagOptionsListView: some View {
+        VStack(alignment: .leading) {
+            ForEach(options) { flag in
+                FlagPickerRow(flag: flag, selection: $selectedOption)
+                BeveledSeparator()
+            }
+        }
+        .background(LinearGradient.cardBackground)
+        .clipShape(RoundedRectangle(cornerRadius: 15))
+    }
+}
+
+/// A single row for a single flag option
+struct FlagPickerRow: View {
+    var flag: FlagOption
+    @Binding var selection: FlagOption?
+
+    var isSelected: Bool {
+        selection?.id == flag.id
+    }
+
+    var body: some View {
+        HStack(alignment: .center) {
+            VStack(alignment: .leading, spacing: 8) {
```

Comment:
It looks like the Spam one isn't quite centered vertically and needs to move down a bit -- probably because there's no description for it. I can't tell what we'd need to do to fix that just by looking, but I'd be happy to help debug if you'd like.